applications/nrf5340_audio/src/bluetooth/bt_content_control/media/bt_content_ctrl_media.c

```diff
@@ -0,0 +1,495 @@
+		LOG_ERR("MCC: Media State read failed (%d)", err);
+		return;
+	}
+
+	media_player_state = state;
+}
+
+/**
+ * @brief  Callback handler for received MCS commands.
+ *
+ * @note	This callback will be triggered when the server has received a
+ *		command from the client or the commander.
+ */
+static void mcs_command_recv_cb(struct media_player *plr, int err,
+				const struct mpl_cmd_ntf *cmd_ntf)
+{
+	if (err) {
+		LOG_ERR("Command failed (%d)", err);
+		return;
+	}
+
+	LOG_DBG("Received opcode: %d", cmd_ntf->requested_opcode);
+
+	if (cmd_ntf->requested_opcode == BT_MCS_OPC_PLAY) {
+		play_pause_cb(true);
+	} else if (cmd_ntf->requested_opcode == BT_MCS_OPC_PAUSE) {
+		play_pause_cb(false);
+	} else {
+		LOG_WRN("Unsupported opcode: %d", cmd_ntf->requested_opcode);
+	}
+}
+
+/**
+ * @brief  Callback handler for getting the current state of the media player.
+ *
+ * @note	This callback will be triggered when the server has asked for the
+ *		current state of its local media player.
+ */
+static void mcs_media_state_cb(struct media_player *plr, int err, uint8_t state)
+{
+	if (err) {
+		LOG_ERR("Media state failed (%d)", err);
+		return;
+	}
+
+	media_player_state = state;
+}
+
+/**
+ * @brief  Callback handler for getting a pointer to the local media player.
+ *
+ * @note	This callback will be triggered during initialization when the
+ *		local media player is ready.
+ */
+static void mcs_local_player_instance_cb(struct media_player *player, int err)
+{
+	struct mpl_cmd cmd;
+
+	if (err) {
+		LOG_ERR("Local player instance failed (%d)", err);
+		return;
+	}
+
+	LOG_DBG("Received local player");
+
+	local_player = player;
+
+	cmd.opcode = BT_MCS_OPC_PLAY;
+
+	/* Since the media player is default paused when initialized, we
+	 * send a play command when the first stream is enabled
+	 */
+	(void)media_proxy_ctrl_send_command(local_player, &cmd);
+}
+
+/**
+ * @brief	Send command to either local media player or peer
+ *
+ * @param	conn	Pointer to the conn to send the command to
+ * @param	cmd	Command to send
+ *
+ * @return	0 for success, error otherwise.
+ */
+static int mpl_cmd_send(struct bt_conn *conn, struct mpl_cmd *cmd)
+{
+	int ret;
+	int failed_ret = 0;
+
+	if (IS_ENABLED(CONFIG_BT_MCS)) {
+		ret = media_proxy_ctrl_send_command(local_player, cmd);
+		if (ret) {
+			LOG_WRN("Failed to send command: %d", ret);
+			return ret;
+		}
+	}
+
+	if (IS_ENABLED(CONFIG_BT_MCC)) {
+		if (conn != NULL) {
+			int idx = mcc_peer_index_get(conn);
+
+			if (idx < 0) {
+				LOG_ERR("Unable to find mcc_peer");
+				return idx;
+			}
+
+			if (mcc_peer[idx].mcp_mcs_disc_status == FINISHED) {
+				ret = bt_mcc_send_cmd(mcc_peer[idx].conn, cmd);
+				if (ret) {
+					LOG_WRN("Failed to send command: %d", ret);
+					return ret;
+				}
+			} else {
+				LOG_WRN("MCS discovery has not finished: %d",
+					mcc_peer[idx].mcp_mcs_disc_status);
+				return -EBUSY;
+			}
+
+			return 0;
+		}
+
+		/* Send cmd to all peers connected and has finished discovery */
```

Comment:
Correct, in the current application ``conn`` will be ``NULL`` most of the times this is called, since we don't have a specific ``conn`` to do play/pause for when the button is pressed.